content/developer/reference/guidelines.rst

```diff
@@ -999,17 +999,88 @@ More precise JS guidelines are detailed in the `github wiki  <https://github.com
 You may also have a look at existing API in Javascript by looking Javascript
 References.
 
-CSS coding guidelines
----------------------
+Template guidelines
+-------------------
+
+Structure
+~~~~~~~~~
+
+.. note:: These rules are in place because the general structure of the template is considered more important than the details of each line.
+
+- Each block element should be on its own line.
+- Small inline/style/text elements can be placed either on the same line as their parent or on their own line (depending on their importance and size).
+- Indentation of 4 spaces is applied between a parent and its children elements. The closing tag of a parent should always be at the same indentation level as its opening tag.
+- It is preferable to use instructions (such as `t-if`) directly on the element on which they apply (as opposed to having them on a separate `<t>` element).
```

Comment:
There's a rationale for this. Extending a template behaves differently whether the original template has a `<t t-name>` or (for instance) `<div t-name>` root element. In the later case, the extension may replace the root node tag.

Reply:
For the `t-name` I agree, it is not part of the guideline but it should be added. The template definition must be on its own `<t>` tag.

Reply:
For other `t-` instructions, our current code actually does like you suggest and like you did too, which is to have them separately. But this approach of having the conditions standalone make the template look like code/instructions instead of like a declarative render.

The goal of the guideline is to ensure a flatter HTML, and have sibling elements in the result actually be sibling in the template. The fact that some elements are shown or not depending on some conditions is actually a detail of themselves, hence the proposition to have them as "attribute" instead of as "structure".

Reply:
I don't see how it is more declarative to put the t-if on the tag itself. Also, I could argue that there are obvious downsides: it also crowds the list of attributes on that tag, and make it potentially less clear. Multiple directives on the same node have different priorities, which is made obvious if the t-if is separated on a parent t node.  

In any case, in my opinion, I don't think either solution is best for all cases, so I think that it is better to leave that one out of the guideline, otherwise we will lose so much time arguing about it later (see FP comment on that).